Exercise10Keenan.R

```diff
@@ -0,0 +1,88 @@
+#Problem 1: Cumulative score graphing
+library(ggplot2)
+#Load in data
+rawData <- read.table('UWvMSU_1-22-13.txt', header = TRUE)
+
+#create cumulative score variables and dataframes
+UWCumScore <- 0
+MSUCumScore <- 0
+uw <- data.frame(matrix(ncol = 2, nrow = 0))
+msu <- data.frame(matrix(ncol = 2, nrow = 0))
+first_entry <- data.frame(0,0)
+names(first_entry)<-c('time', 'score')
+cn <- c("time", "score")
+colnames(uw) <- cn
+colnames(msu) <- cn
+#put in start score of 0-0
+uw<-rbind(uw, first_entry)
+msu<-rbind(msu, first_entry)
+
+#populate with cumulative score at each time scored.
+for (i in 1:nrow(rawData)) {
+  if (rawData[i,]$team =="UW"){
+    UWCumScore <- UWCumScore + rawData[i,]$score
+    addition1 <- data.frame(rawData[i,]$time, UWCumScore)
+    addition2 <- data.frame(rawData[i,]$time, MSUCumScore)
+    names(addition1)<- c('time', 'score')
+    names(addition2)<- c('time', 'score')
+    uw<-rbind(uw, addition1)
+    msu<-rbind(msu, addition2)
+  } 
+  else{
+    MSUCumScore <- MSUCumScore + rawData[i,]$score
+    addition1 <- data.frame(rawData[i,]$time, MSUCumScore)
+    addition2 <- data.frame(rawData[i,]$time, UWCumScore)
+    names(addition1)<- c('time', 'score')
+    names(addition2)<- c('time', 'score')
+    msu<-rbind(msu, addition1)
+    uw<-rbind(uw, addition2)
+  }
+}
+
+#add final score entries
+final_entry_uw <- data.frame(40,UWCumScore)
+final_entry_msu <- data.frame(40, MSUCumScore)
+names(final_entry_msu)<- c('time', 'score')
+names(final_entry_uw)<- c('time', 'score')
+uw<-rbind(uw, final_entry_uw)
+msu<-rbind(msu, final_entry_msu)
+

```

Comment:
A different approach that is more efficient. You don't need to use rbind every time:
# load table of scoring
scoring=read.table("UWvMSU_1-22-13.txt",header=TRUE,sep="\t",stringsAsFactors=FALSE)

# look at data
dim(scoring)
head(scoring)

# preallocating matrix to store cumulative scores
cum_scores=matrix(NA,nrow(scoring)+1,3)
cum_scores[,1]=c(0,scoring[,1])
cum_scores[1,2:3]=0
colnames(cum_scores)=c("time","UW","MSU")

# looping through individual scoring events
for(i in 1:nrow(scoring)){
  if(scoring[i,2]=="UW"){
    cum_scores[(i+1),2]=cum_scores[i,2]+scoring[i,3]
    cum_scores[(i+1),3]=cum_scores[i,3]
  }else{
    cum_scores[(i+1),2]=cum_scores[i,2]
    cum_scores[(i+1),3]=cum_scores[i,3]+scoring[i,3]
  }
}